src/routes/auth/user.ts

```diff
@@ -23,77 +23,89 @@ export interface UserLoginData {
+    if (userUsernameDoc) return res.status(400).json({ username: "this username is already taken" });
+    if (userEmailDoc) return res.status(400).json({ email: "this email is already taken" });
+
+    // Gen salt and store it together with the hashed password
+    const userPasswordSalt = await bcrypt.genSalt(10);
+    const hashedPassword = <string>await bcrypt.hash(newUser.password, userPasswordSalt);
+
+    // Add user to DB
+    const _newUserModel = (
+      await UserModel.create({
+        email: newUser.email,
+        username: newUser.username,
+        password: hashedPassword,
+        password_salt: userPasswordSalt,
+        private_profile: false,
+        banned: false,
+      })
+    ).save();
+
+    // Return auth token;
+    let accessToken;
+    const TOKEN = process.env.ACCESS_TOKEN_SECRET;
+    TOKEN ? (accessToken = jwt.sign({ email: newUser.email, username: newUser.username }, TOKEN)) : console.log("No secret access token in env");
+
+    res.status(200).json({ jwt: `Bearer ${accessToken}` });
+  } catch (err) {
+    console.log(err);
+  }
 });
 
 //user login route
 userAuthRouter.post("/login", async (req, res) => {
-  const userLoginData: UserLoginData = { email: req.body.email, password: req.body.password };
+  try {
+    const userLoginData: UserLoginData = { email: req.body.email, password: req.body.password };
 
-  const { valid, errors } = validateUserLoginData(userLoginData);
+    const { valid, errors } = validateUserLoginData(userLoginData);
 
-  if (!valid) return res.status(400).json(errors);
+    if (!valid) return res.status(400).json(errors);
 
-  // Authenticate user
+    // Authenticate user
 
-  // checking if user exists in DB
-  const userEmailDoc = await UserModel.findOne({ email: userLoginData.email }).exec();
-  if (!userEmailDoc) return res.status(400).json({ login: "email or password is incorrect" });
-  else if (!(await userEmailDoc.checkPassword(userLoginData.password))) {
-    return res.status(400).json({ login: "email or password is incorrect" });
-  }
+    // checking if user exists in DB
+    const userEmailDoc = await UserModel.findOne({ email: userLoginData.email }).exec();
+    if (!userEmailDoc) return res.status(400).json({ login: "email or password is incorrect" });
+    else if (!(await userEmailDoc.checkPassword(userLoginData.password))) {
+      return res.status(400).json({ login: "email or password is incorrect" });
+    }
 
-  // Return auth token;
-  let accessToken;
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
-  TOKEN ? (accessToken = jwt.sign({ email: userEmailDoc.email, username: userEmailDoc.username }, TOKEN)) : console.log("No secret access token in env");
+    // Return auth token;
+    let accessToken;
+    const TOKEN = process.env.ACCESS_TOKEN_SECRET;
+    TOKEN ? (accessToken = jwt.sign({ email: userEmailDoc.email, username: userEmailDoc.username }, TOKEN)) : console.log("No secret access token in env");
```

Comment:
With a proper typing file (*.d.ts) you won't have to check for existence of env variable (since we use dotenv-safe).